pkg/controller/operconfig/operconfig_controller.go

```diff
@@ -264,15 +268,22 @@ func (r *ReconcileOperConfig) Reconcile(ctx context.Context, request reconcile.R
 	}
 	objs = append([]*uns.Unstructured{app}, objs...)
 
-	// Set up the Pod reconciler before we start creating DaemonSets/Deployments
+	// Set up the Pod reconciler before we start creating DaemonSets/Deployments/StatefulSets
 	daemonSets := []types.NamespacedName{}
 	deployments := []types.NamespacedName{}
+	statefulSets := []types.NamespacedName{}
 	relatedObjects := []configv1.ObjectReference{}
 	for _, obj := range objs {
 		if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "DaemonSet" {
 			daemonSets = append(daemonSets, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
 		} else if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "Deployment" {
 			deployments = append(deployments, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
+		} else if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "StatefulSet" {
+			statefulSets = append(statefulSets, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
+			continue
```

Comment:
is that `continue` intended? I am not sure why we need that here

Reply:
The `deleteRelatedObjectsNotRendered` in status manager doesn't support deleting resources in mgmt, thus I skip adding it to relatedObjects. I will fix it in the next push.

Reply:
I mean that the `continue` skips when an object is a `StatefulSet`, it shouldn't just based on that, right?

Reply:
removed `continue`.